### DISCUSSION TITLE
How does `mixHash` be used in qbft consensus?

### DISCUSSION THREAD

**Discussion by kth-tw** [ConsenSys/quorum#1392] [Q&A] [answered]
I tried to use `quorum-genesis-tool` build a quorum network. The genesis.json include a value `mixHash`, ant it is always set to `0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365` in [here](https://github.com/ConsenSys/quorum-genesis-tool/blob/2bb998b3419557429f4731a34017a54b8dd57b0d/src/lib/genesisGenerate.ts#L42)
If I choose `qbft` as consensus, what does this value mean to?

**Comment by baptiste-b-pegasys** [comment, marked as answer]
Hello, from our code, this value is the istanbul digest.
```go
	// IstanbulDigest represents a hash of "Istanbul practical byzantine fault tolerance"
	// to identify whether the block is from Istanbul consensus engine
	IstanbulDigest = common.HexToHash("0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365")
```

In ethereum, mixHash is used for the mining PoW consensus (https://gist.github.com/0mkara/b953cc2585b18ee098cd#file-genesis-md), for our PoA, it's for identifying the consensus engine, that's the same function.

You may try a different value and see if it is possible to change it as you want to.

**Reply by kth-tw** [reply]
I guess `mixHash` is used [here](https://github.com/ConsenSys/quorum/blob/e95f163de1d2f00c06693a7f9f5955b4ffe9fa94/core/types/block.go#L101-L108)

If I am interested in this part(genesis file, istanbul digest, ...etc) , is there a document to refer to?

**Reply by baptiste-b-pegasys** [reply]
Ok therefore it's not possible to change the mixHash for istanbul consensus.
I found https://github.com/ethereum/EIPs/issues/650 (mixHash: Fixed magic number 0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365 for Istanbul block identification.)

https://github.com/ConsenSys/quorum/issues/1084

**Reply by kth-tw** [reply]
Thank you for your help finding reference.  It seems I had a big misunderstood.  This is the parameter of geth, not added by quorum.
